src/Symfony/Component/HttpKernel/Bundle/SimpleBundle.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Bundle;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * An implementation of a simple all-in-one {@link BundleInterface} and
+ * configurable {@link Extension}.
+ *
+ * @author Guilhem N. <egetick@gmail.com>
+ */
+abstract class SimpleBundle extends Bundle implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public function getContainerExtension()
+    {
+        if (null === $this->extension) {
+            $this->extension = new SimpleExtension($this);
+        }
+
+        return $this->extension;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root($this->getContainerExtension()->getAlias());
+
+        $this->buildConfiguration($rootNode);
+
+        return $treeBuilder;
+    }
+
+    protected function buildConfiguration(TreeBuilder $rootNode)
+    {
+    }
+
+    protected function buildContainer(array $config, ContainerBuilder $container)
+    {
+    }
+}
+
+/**
+ * Simple {@link Extension} supporting {@link SimpleBundle}.
+ *
+ * @internal
+ */
+final class SimpleExtension extends Extension
+{
+    private $bundle;
+
+    public function __construct(SimpleBundle $bundle)
+    {
+        $this->bundle = $bundle;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAlias()
+    {
+        // check naming convention
+        $basename = preg_replace('/Bundle$/', '', $this->bundle->getName());
+
+        return Container::underscore($basename);
```

Comment:
No problem, it's made to be reused :P
I think it loses of its interest if not in symfony, let's wait to see what happens ☺